### DISCUSSION TITLE
OCR in place?

### DISCUSSION THREAD

**Discussion by gerroon** [ocrmypdf/OCRmyPDF#1224] [Q&A] [answered]
Hi

Is it possible to OCR a document and replace it without providing an output.pdf type usage? I want to use "find" on the Linux terminal but it gets very ugly to use a temporary file with it. I want to basically overwrite existing Calibre library with PDFs.

thanks

**Comment by jbarlow83** [comment, marked as answer]
Yes `ocrmypdf x.pdf x.pdf ` will overwrite in place on success and leave the file untouched on failure. There's a bunch of information and example scripts in the documentation on batch processing that may be helpful too.
